File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function changeNum(num, base) {
    let res ='';
    
    num = parseInt(num);
    
    while(num > 0) {
        res = num % base + res;
        
        num = Math.floor(num/base);
    }
    
    return res;
}

function solution(n, k) {
    var answer = 0;
    
    const num = changeNum(n, k);
    
    let nums = [];
    let tmp = '';
    for(const item of num) {
        if(item === '0') {
            if(tmp !== '') {
                nums.push(parseInt(tmp));
                tmp = '';
            }
            continue;
        }
        
        tmp += item;
    }
    
    if(tmp !== '') nums.push(parseInt(tmp));
    
    for(const item of nums) {
        
        if(item === 1) continue;
        
        let flag = true;
        const len = Math.sqrt(item,2);
        for(let i = 2; i<=len; i++ ) {
            if(item % i === 0) {
                flag = false;
                break;
            }
        }
        
        if(flag) answer +=1;
    }
    
    
    
    return answer;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

string changeNum(int num, int base)
{
    string res = "";
    
    while(num >0)
    {
        res = char('0' + num%base) + res;
        
        num = num/base;
    }
    
    return res;
}

int solution(int n, int k) {
    int answer = 0;
    
    string str = changeNum(n, k);
    
    vector<long long> nums;
    string word = "";
    for(char& c : str)
    {
        if(c == '0' )
        {
            if(word != "")
            {
                nums.push_back(stoll(word));
                word = "";
            }
            continue;
        }
        
        word += c;
    }
    
    if(word != "") nums.push_back(stoll(word));
    
    for(auto& item : nums)
    {
        if(item == 1) continue;
        
        long long len = sqrt(item);
        bool flag = true;
        
        for(long long i=2; i<= len ;i++)
        {
            if(item % i ==0) {
                flag = false;
                break;
            }
        }
        
        if(flag) answer +=1;
    }
    
    
    return answer;
}